Rover Docs Rough Pass (#160)
apollographql/apollo-cli: docs/source/graphs.md

graph.

> Everything on this page applies only to non-federated graphs. For a guide on
> how to work with a federated graph, check out [Working with Subgraphs](./subgraphs).

## Introspecting Schemas

Sometimes, you don't have a `.graphql` SDL file readily available. This is
especially true with graphs built using certain libraries which don't use SDL
to define schemas at all like `graphql-kotlin`.

For cases like this, or to download a schema from a publicly available endpoint,
try using Rover's introspection command:

```bash
rover graph introspect http://my-public-api.com
```

> TODO: headers API

Some endpoints will require certain headers to be passed...

## Pushing a schema to Apollo Studio

> All of the following examples require you to first create an account with
> Apollo Studio, [authenticate Rover]
> (./configuring#authenticating-with-apollo-studio) and create a new graph to
> publish to.

Apollo Studio is a cloud platform that helps you build, validate, and secure
your organization's data graph. The Apollo schema registry, as part of Apollo
Studio is a free tool to allow you to track schema versions and validate

> changes, all while acting as the single source of truth for the question of
> "what schema is the correct one?"

Reviewer: ```suggestion
changes, all while acting as the single source of truth for your graph's schema(s).
```

Rover provides multiple convenient ways of publishing schemas. If you're using
a `.graphql` SDL file, you can pass the path to that file as the `--schema` flag
(or `-s` for short):

```bash
rover graph push my-graph --schema ./schema.graphql
```

> this will push the schema at ./schema.graphql to the default (`current`) variant

Reviewer: ```suggestion
This will push the schema at ./schema.graphql to the default (`current`) variant
```
